plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/fileio/ForwardingFileIo.java

```diff
@@ -31,18 +33,40 @@
 import java.util.Map;
 import java.util.stream.Stream;
 
+import static io.airlift.units.DataSize.Unit.MEGABYTE;
 import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.TimeUnit.MINUTES;
 import static java.util.stream.Collectors.joining;
+import static org.apache.iceberg.CatalogProperties.CACHE_EXPIRATION_INTERVAL_MS;
+import static org.apache.iceberg.CatalogProperties.IO_MANIFEST_CACHE_ENABLED;
+import static org.apache.iceberg.CatalogProperties.IO_MANIFEST_CACHE_MAX_TOTAL_BYTES;
+import static org.apache.iceberg.ManifestFiles.dropCache;
 
 public class ForwardingFileIo
         implements SupportsBulkOperations
 {
     private static final int DELETE_BATCH_SIZE = 1000;
     private static final int BATCH_DELETE_PATHS_MESSAGE_LIMIT = 5;
+    private static final String PER_QUERY_MANIFEST_FILE_CACHE_SIZE = String.valueOf(DataSize.of(40, MEGABYTE).toBytes());
```

Comment:
Add configurable properties

Reply:
Added `iceberg.per-query-manifest-cache.max-size`. Didn't make the TTL configurable as that's just an internal thing we're setting to ensure clean up of cache, from user perspective the cache TTL is the duration of the query.